device-registry/src/main/scala/org/genivi/sota/device_registry/DevicesResource.scala

```diff
@@ -140,15 +148,20 @@ class DevicesResource(namespaceExtractor: Directive1[AuthedNamespaceScope],
   }
 
   def mydeviceRoutes: Route = namespaceExtractor { authedNs =>
-    (pathPrefix("mydevice") & extractUuid) { uuid =>
-      (post & path("ping") & authedNs.oauthScope(s"ota-core.${uuid.show}.write")) {
-        updateLastSeen(uuid)
-      } ~
-      (get & pathEnd & authedNs.oauthScopeReadonly(s"ota-core.${uuid.show}.read")) {
-        fetchDevice(uuid)
+    pathPrefix("mydevice") {
+      extractUuid { uuid =>
+        (post & path("ping") & authedNs.oauthScope(s"ota-core.${uuid.show}.write")) {
+          updateLastSeen(uuid)
+        } ~
+        (get & pathEnd & authedNs.oauthScopeReadonly(s"ota-core.${uuid.show}.read")) {
+          fetchDevice(uuid)
+        }
+        (put & path("packages") & authedNs.oauthScope(s"ota-core.{device.show}.write")) {
+           updateInstalledSoftware(uuid)
+        }
       } ~
       (put & path("packages") & authedNs.oauthScope(s"ota-core.{device.show}.write")) {
-        updateInstalledSoftware(uuid)
+        resolverPackageMigration()
```

Comment:
It doesn't come from a device, true. However, InstalledPackages on device registry has no namespace field, so a namespace for this data doesn't really make sense. Putting it in mydevices was an attempt to bypass authentication as much as possible. I'm open to ideas though.